discovery/marathon/marathon.go

```diff
@@ -140,37 +146,26 @@ func NewDiscovery(conf SDConfig, logger log.Logger) (*Discovery, error) {
 		logger = log.NewNopLogger()
 	}
 
-	tls, err := httputil.NewTLSConfig(conf.TLSConfig)
+	client, err := httputil.NewClientFromConfig(conf.HTTPClientConfig, "marathon_sd")
 	if err != nil {
 		return nil, err
 	}
 
-	token := string(conf.BearerToken)
-	if conf.BearerTokenFile != "" {
-		bf, err := ioutil.ReadFile(conf.BearerTokenFile)
+	authToken := conf.AuthToken
+	if len(authToken) == 0 && len(conf.AuthTokenFile) > 0 {
+		b, err := ioutil.ReadFile(conf.AuthTokenFile)
```

Comment:
This needs to be read on every request.

Reply:
For this part, I did the same as what the http client does for the `bearer_token_file` configuration property [here](https://github.com/prometheus/common/blob/93dfbf7/config/http_config.go#L135) which doesn't seem to read the file on every request.

Perhaps I am mis-reading that code somehow? Or perhaps that change is in the works? I can certainly give it a stab if you feel strongly about supporting this now.

Reply:
That's not right, https://github.com/prometheus/prometheus/pull/3709/files was meant to do that.

Reply:
Oh, ok, I looked at the wrong http client. Alrighty, I'll take a stab at implementing this.

Reply:
You're looking at the right one, it needs fixing.

Reply:
I may need a bit of guidance on this one. The approach I'm considering is to create a `AuthTokenRoundTripper` and then use that to wrap the `RoundTripper` that `httputil` creates. But to put that new `rt` on the client, I see 2 approaches:

* create client using `NewClientFromConfig` and then override its `RoundTripper` member
* create `NewHTTPRoundTripperFromConfig` in `httputil` (which is basically `NewHTTPClientFromConfig` minus the call to `newClient`)

Mutating the client seems like a bad idea to me, but it makes the change entirely local to `marathon.go` which perhaps is what we want in this case, I'm not sure.

Perhaps there's an alternative approach that you might suggest?

Reply:
Ok, I took the second approach in 2a0e760 and 904902e. Please let me know if that's okay.